sdk/storage/azure-storage-blob/v10_v12.md

```diff
@@ -0,0 +1,437 @@
+    .credential(myCredential)
+    .buildBlockBlobAsyncClient(); // we choose the "async" method this time
+``` 
+
+## Credentials
+
+### Connection string 
+v10/v11 did not support connection strings. In v12, our builders accept connection strings, no calls to `endpoint()` or `credential()` required. However, connection strings generally only point at the account, so you WILL need to specify container and blob names as necessary.
+```java
+ContainerClient client = new ContainerClientBuilder()
+    .connectionstring("${connectionstring}")
+    .containerName("mycontainer")
+    .buildBlockBlobClient();
+```
+
+### Shared key credential
+Shared key credentials can be used directly to authenticate your client. You need to fetch the `${accountName}` and `${accountKey}` from Azure Portal.
+Learn More from [README](https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/storage/azure-storage-blob/README.md)
+
+In v10/v11, you used the credential to build your pipeline.
+```java
+// Initialize the credentials
+SharedKeyCredentials credential = new SharedKeyCredentials("${accountName}", "${accountKey}");
+// How service object take in the credentials.
+BlockBlobURL blockBlobURL = new BlockBlobURL(new URL("<your-storage-blob-url>"), StorageURL.createPipeline(credential, new PipelineOptions()));
+```
+
+Since the builder in v12 manages pipeline generation, you can just hand the `SharedKeyCredential` to the builder.
+```java
+// Initialize the credentials
+SharedKeyCredential credential = new SharedKeyCredential("${accountName}", "${accountKey}");
+// How service object take in the credentials.
+BlobServiceClient client = new BlobServiceClientBuilder()
+    .endpoint("https://" + accountName + ".blob.core.windows.net")
+    .credential(credential)
+    .buildBlobServiceClient();
+```
+
+### SASToken
+A URL with a SAS token looks like the following: `https://${accountName}.blob.core.windows.net/?${sasToken}`.
+Refer the [documentation](https://github.com/Azure/azure-sdk-for-java/tree/azure-storage-file_12.0.0-preview.3/sdk/storage/azure-storage-blob/README.md) for more info.
+
+See section [Generate SAS token](#generate-sas-token) for how to generate the token programatically.
+
+In v10/v11, you had to provide the URL with the token attached, which often resulted in URL manipulation when a token could be used on more than one resource.
+```java
+// Here is how the blob object take in the sas token generated in above section.
+BlobURLParts parts = URLParser.parse(cu.createBlobURL(blobName).toURL())
+parts.withSasQueryParameters(sasToken).withScheme("https")
+BlockBlobURL bu = new BlockBlobURL(parts.toURL(), StorageURL.createPipeline(
+    new AnonymousCredentials(), new PipelineOptions()))
+```
+
+In v12, SAS tokens can be managed just like any other credential.
+```java
+// Initialize the credentials
+SASTokenCredential credential = SASTokenCredential.fromSASTokenString("${sasToken}");
```

Comment:
This is outdated. There is no longer a SasTokenCredential type. The builder now just has a sasToken(String) method